allofplos/makedb.py

```diff
@@ -105,26 +104,32 @@ class PLOSArticle(BaseModel):
     word_count = IntegerField()
     JATS_type = ForeignKeyField(JATSType, related_name='jats')
 
+class SubjectsPLOSArticle(BaseModel):
+    subject = ForeignKeyField(Subjects)
+    article = ForeignKeyField(PLOSArticle)
+
 class CoAuthorPLOSArticle(BaseModel):
     corr_author = ForeignKeyField(CorrespondingAuthor)
     article = ForeignKeyField(PLOSArticle)
 
 db.connect()
-db.create_tables([Journal, PLOSArticle, ArticleType,
-                      CoAuthorPLOSArticle, CorrespondingAuthor,
-                      JATSType, Affiliations, Country])
+db.create_tables([Journal, PLOSArticle, ArticleType, CoAuthorPLOSArticle,
+                  CorrespondingAuthor, JATSType, Affiliations, Country,
+                  SubjectsPLOSArticle, Subjects])
+
 
 if args.starter:
-    allfiles = os.listdir('starter_corpus')
+    corpus = Corpus(starterdir)
+    allfiles = corpus.iter_files
 else:
-    allfiles = os.listdir(corpusdir)
+    corpus = Corpus()
```

Comment:
This should still work even if you don't pass in `get_corpus_dir()` because it is the default in `Corpus`, and that should work using `None` as the default value.

Reply:
If it doesn't work, please explain… because that means there's a bug elsewhere.

Reply:
It is working, I am not passing get_corpus_dir() anymore.